api/src/API/Controllers/WildApricotController.cs

```diff
@@ -0,0 +1,119 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using RaceResults.Common.Models;
+using RaceResults.Data.KeyVault;
+using RestSharp;
+
+namespace RaceResults.Api.Controllers
+{
+    [ApiController]
+    [Route("wa")]
+    public class WildApricotController : ControllerBase
+    {
+        public static string Base64Encode(string plainText)
+        {
+            var plainTextBytes = System.Text.Encoding.UTF8.GetBytes(plainText);
+            return System.Convert.ToBase64String(plainTextBytes);
+        }
+
+        // FOR TESTING. Need to figure out something better...
+        public static bool OverrideAuth { get; set; }
+
+        public static async Task<bool> Authorized(HttpRequest request, string orgAssignedMemberId)
+        {
+            if (OverrideAuth)
+            {
+                return true;
+            }
+
+            var hasAccount = int.TryParse(request.Headers["WA-AccountId"], out var accountId);
+            var authorization = request.Headers["WA-Authorization"];
+
+            if (!hasAccount || string.IsNullOrEmpty(authorization))
+            {
+                return false;
+            }
+
+            var waResponse = await WildApricotController.GetCurrentUserFromWildApricot(accountId, authorization);
+            if (waResponse.id.ToString() != orgAssignedMemberId)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public static async Task<WildApricotMember> GetCurrentUserFromWildApricot(int accountId, string authorization)
+        {
+            var client = new RestClient("https://api.wildapricot.org/publicview/v1");
+
+            var request = new RestRequest($"accounts/{accountId}/contacts/me?includeDetails=false", DataFormat.Json);
+            request.AddHeader("Authorization", authorization);
+
+            return await client.GetAsync<WildApricotMember>(request);
+        }
+
+        private readonly ILogger<OrganizationsController> logger;
+
+        private readonly IKeyVaultClient keyVaultClient;
+
+        public WildApricotController(
+                IKeyVaultClient keyVaultClient,
+                ILogger<OrganizationsController> logger)
+        {
+            this.keyVaultClient = keyVaultClient;
+            this.logger = logger;
+        }
+
+        public class GetAccessTokenBody
+        {
+            public string OrganizationId { get; set; }
+
+            public string ClientId { get; set; }
+
+            public string RedirectURI { get; set; }
+
+            public string Scope { get; set; }
+        }
+
+        [HttpPost("oauth/{authorization_code}")]
+        public async Task<IActionResult> GetAccessToken(string authorization_code, [FromBody] GetAccessTokenBody body)
+        {
+            var secretName = body.OrganizationId + "-client-secret";
+            var clientSecret = await this.keyVaultClient.GetSecretAsync(secretName);
+
+            using (var client = new HttpClient())
```

Comment:
HttpClients should be instantiated once and reused throughout the life of the application. Having a new one created on every call to this unauthenticated endpoint can result in running out of available sockets.

Reply:
Can you fix this? I'm not familiar with how to do this properly